### PR TITLE
samples: matter: added dfu support to the template app

### PR DESCRIPTION
**Pull request by markaj-nordic**
DFU is mandatory for every Matter device so it makes sense to enable this feature for all matter samples, even template application.

Changes verified with Linux ota-provider (announcing lock app image) and nrf52840dk_nrf52840 build with DFU enabled.

### CODE REVIEW

File: samples/matter/template/README.rst [hunk 1]
```diff
@@ -80,11 +80,41 @@ This sample supports the following build types, depending on the selected board:
...
+* ``no_dfu`` -- Debug version of the application without Device Firmware Upgrade feature support - can be used only for the nRF52840 DK and nRF5340 DK, as those platforms have DFU enabled by default.
 
 .. note::
     `Selecting a build type`_ is optional.
     The ``debug`` build type is used by default if no build type is explicitly selected.
 
+Device Firmware Upgrade support
+===============================
+
+.. matter_template_sample_build_with_dfu_start
```
Comment (kkasperczyk-no) [inline samples/matter/template/README.rst:92]: Can't we just add:

.. include:: ../lock/README.rst
    :start-after: matter_door_lock_sample_build_with_dfu_start
    :end-before: matter_door_lock_sample_build_with_dfu_end

?

**Review by kkasperczyk-no — COMMENTED** [review]
Could you align to the changes done in: https://github.com/nrfconnect/sdk-nrf/pull/7252 ?

**Comment by markaj-nordic** [comment]
Fixed formatting and commit message compliance.

File: samples/matter/template/README.rst (hunk 1 above)
Comment (markaj-nordic) [reply samples/matter/template/README.rst:92]: Sounds good :)

**Comment by markaj-nordic** [comment]
> Could you align to the changes done in: #7252 ?

@kkasperczyk-no done.

**Review by kkasperczyk-no — APPROVED** [review]

**Comment by markaj-nordic** [comment]
Rebased and fixed conflicts.

**Review by LuDuda — APPROVED** [review]

**Review by greg-fer — APPROVED** [review]
I guess we need a changelog entry for this kind of changes, no?

**Comment by kkasperczyk-no** [comment]
Could you add no_dfu configuration to the sample.yaml to test also this build in twister?

**Pull request merged**